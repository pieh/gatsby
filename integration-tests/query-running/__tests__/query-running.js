const execa = require(`execa`)
const { resolve } = require(`path`)

const {
  clearTestOutput,
  getTestInputs,
  readTestOutput,
} = require(`../test-utils`)

const rootPath = resolve(`${__dirname}/..`)

describe(`Query running`, () => {
  const dbEngines = [`redux`, `loki`]

  clearTestOutput()
  const testInputs = getTestInputs()

  // do both builds first
  dbEngines.forEach(dbEngine => {
    execa.shellSync(`yarn clean-and-build`, {
      cwd: rootPath,
      env: { GATSBY_DB_NODES: dbEngine },
    })
  })

  testInputs.forEach(testInput => {
    describe(testInput.name, () => {
      const [reduxResults, lokiResults] = dbEngines.map(dbEngine =>
        readTestOutput(testInput, dbEngine)
      )

      it(`Using redux matches snapshot`, () => {
        expect(reduxResults).toMatchSnapshot()
      })
      it(`Using loki matches redux`, () => {
        expect(lokiResults).toEqual(reduxResults)
      })
    })
  })
})
