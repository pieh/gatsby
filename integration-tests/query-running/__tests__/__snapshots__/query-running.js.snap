// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query running basic connection Using redux matches snapshot 1`] = `
Object {
  "data": Object {
    "allEntry": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "entry_1",
            "string": "a",
          },
        },
        Object {
          "node": Object {
            "id": "entry_2",
            "string": "n",
          },
        },
        Object {
          "node": Object {
            "id": "entry_3",
            "string": "c",
          },
        },
        Object {
          "node": Object {
            "id": "entry_4",
            "string": "d",
          },
        },
      ],
    },
  },
}
`;

exports[`Query running basic sort Using redux matches snapshot 1`] = `
Object {
  "data": Object {
    "allEntry": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "entry_1",
            "sortField": "a",
            "string": "a",
          },
        },
        Object {
          "node": Object {
            "id": "entry_4",
            "sortField": "a",
            "string": "d",
          },
        },
        Object {
          "node": Object {
            "id": "entry_2",
            "sortField": null,
            "string": "n",
          },
        },
        Object {
          "node": Object {
            "id": "entry_3",
            "sortField": null,
            "string": "c",
          },
        },
      ],
    },
  },
}
`;
